OrbitBase/include/OrbitBase/Path.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_PATH_H_
+#define ORBIT_BASE_PATH_H_
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Result.h"
+
+namespace OrbitBase {
+[[nodiscard]] std::string GetExecutablePath();
+[[nodiscard]] std::string GetExecutableDir();
+[[nodiscard]] std::string GetFileName(const std::string& file_path);
+[[nodiscard]] std::string StripExtension(const std::string& file_path);
+[[nodiscard]] std::string GetExtension(const std::string& file_path);
+[[nodiscard]] std::string GetDirectory(const std::string& any_path);
+[[nodiscard]] std::string JoinPath(const std::vector<std::string>& parts);
+
+[[nodiscard]] std::vector<std::string> ListFiles(
+    const std::string& directory, const std::function<bool(const std::string&)>& filter =
+                                      [](const std::string&) { return true; });
```

Comment:
As this is `OrbitBase`, some small documentation would be beneficial. For example: does `GetExtension` include the `.`? Does `StripExtension` return the full path or only the file name without extension?